Branch-SDK/src/io/branch/referral/util/PromoViewHandler.java

```diff
@@ -0,0 +1,290 @@
+            layoutParams.setMargins(40, 80, 40, 80);
+
+            layout.addView(webView, layoutParams);
+            layout.setBackgroundColor(Color.parseColor("#11FEFEFE"));
+
+            final TextView confirmTxt = new TextView(currentActivity);
+            confirmTxt.setVisibility(View.GONE);
+            confirmTxt.setBackgroundColor(Color.RED);
+            confirmTxt.setText("Confirm");
+            confirmTxt.setGravity(Gravity.CENTER);
+            confirmTxt.setTextAppearance(currentActivity, android.R.style.TextAppearance_Large);
+            RelativeLayout.LayoutParams txtViewLayoutParams = new RelativeLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
+            txtViewLayoutParams.addRule(RelativeLayout.ALIGN_PARENT_BOTTOM);
+            txtViewLayoutParams.setMargins(40, 0, 40, 30);
+            confirmTxt.setPadding(30, 30, 30, 30);
+            layout.addView(confirmTxt, txtViewLayoutParams);
+
+            final Dialog dialog = new Dialog(currentActivity, android.R.style.Theme_Black_NoTitleBar_Fullscreen);
+            dialog.setContentView(layout);
+            dialog.show();
+            if (callback != null) {
+                callback.onPromoViewVisible(promoView.promoAction_);
+            }
+
+            if (Build.VERSION.SDK_INT >= 19) {
+                webView.setLayerType(View.LAYER_TYPE_HARDWARE, null);
+            }
+
+
+            webView.loadUrl(promoView.webViewUrl_);
+            webView.setBackgroundColor(Color.parseColor("#00FE0000"));
+
+
+            webView.setWebViewClient(new WebViewClient() {
+                @Override
+                public void onPageFinished(WebView view, String url) {
+                    super.onPageFinished(view, url);
+                    layout.setVisibility(View.VISIBLE);
+                    confirmTxt.setVisibility(View.VISIBLE);
+                    view.setVisibility(View.VISIBLE);
+                    dialog.show();
+                    showViewWithAlphaTweening(layout);
+                    showViewWithAlphaTweening(confirmTxt);
+                    showViewWithAlphaTweening(view);
+                }
+            });
+
+            dialog.setOnDismissListener(new DialogInterface.OnDismissListener() {
+                @Override
+                public void onDismiss(DialogInterface dialog) {
+                    isPromoDialogShowing = false;
+                    if (callback != null) {
+                        callback.onPromoViewDismissed(promoView.promoAction_);
+                    }
+                }
+            });
+
+            confirmTxt.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    dialog.dismiss();
+                }
+            });
+        }
+    }
+
+    public void saveAppPromoViews() {
+        JSONArray promoViewArray = Branch.getInstance().getPromoViewData();
```

Comment:
Removed. SDK has no test code  now. Test Json is passed form Test bed as debug deeplink param